### DISCUSSION TITLE
export all assets to a csv file

### DISCUSSION THREAD

**Discussion by umarkhan99** [netbox-community/netbox#9353] [Q&A] [answered]
Is there a way to export all the assets currently sitting in my netbox server to a csv file?

**Comment by bogdancordos** [comment, marked as answer]
I don't think there is possible to export a whole database to a csv, only one table at the time.

**Reply by umarkhan99** [reply]
how can I export one table at a time?

**Reply by bogdancordos** [reply]
The easiest way is to use the "Export" option within your NetBox server:
![image](https://user-images.githubusercontent.com/57483807/168586097-97c3b5ef-1edc-427a-aba7-ae9613a7207a.png)
 And do the same for each view that you need exporting.

The other way is to do it from the database itself (https://www.postgresqltutorial.com/postgresql-tutorial/export-postgresql-table-to-csv-file/)

**Comment by mtinberg** [comment]
Click the Export button on nearly every page.  Be advised though this isn't intended as a backup system to restore into another Netbox, it's a reporting system to make it easy to pull data into a spreadsheet or import from a spreadsheet.

I'm not sure if there is a Netbox to Netbox sync script that runs over the API for setting up QA environments and whatnot but that would be useful to have.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Reply by candlerb** [reply]
> I'm not sure if there is a Netbox to Netbox sync script that runs over the API for setting up QA environments and whatnot but that would be useful to have.

It would be difficult to write, if not impossible without specific knowledge of your environment, because the database IDs of corresponding objects in different Netboxes would be different.

Adding a custom field to all objects containing a UUID would be one general way to approach the problem.  Another would be for the replication system to keep its own database which remembers how to relate the database IDs in system A to the database IDs in system B.

Otherwise, most objects don't have a guaranteed unique key.  There are "slugs", but they are not required to be used (and there was a proposal to remove them entirely: #8036).  Devices don't have to have names at all, and names can duplicate in different sites.  IP addresses can have duplicates.  And so on.
